### PR TITLE
SDK-1703: Supplementary Documents

### PR DESCRIPTION
**Pull request by davidgrayston**
### Added
- Supplementary document support

### Deprecated
- `TaskResponse.getGeneratedTextDataChecks()` this method is now implemented on subclasses

### CODE REVIEW

File: examples/doc-scan/src/controllers/index.controller.js [hunk 1]
```diff
@@ -46,10 +50,15 @@ async function createSession() {
     )
     .withRequestedTask(
       new RequestedTextExtractionTaskBuilder()
-        .withManualCheckNever()
+        .withManualCheckAlways()
```
Comment (echarrod) [inline examples/doc-scan/src/controllers/index.controller.js:53]: Do we want to change this?

File: src/doc_scan_service/session/create/filters/required.supplementary.document.builder.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const Validation = require('../../../../yoti_common/validation');
+const Objective = require('./objective/objective');
+
+class RequiredSupplementaryDocumentBuilder {
+  /**
+   * @param {string[]} countryCodes
+   *
+   * @returns {this}
+   */
+  withCountryCodes(countryCodes) {
```
Comment (echarrod) [inline src/doc_scan_service/session/create/filters/required.supplementary.document.builder.js:13]: Do we have the singular WithCountryCode & WithDocumentType for DocumentFilters? If so, I think we should add here too

File: examples/doc-scan/src/controllers/index.controller.js (hunk 1 above)
Comment (davidgrayston) [reply examples/doc-scan/src/controllers/index.controller.js:53]: This was intentional to help demonstrate checks in the response

File: src/doc_scan_service/session/create/filters/required.supplementary.document.builder.js (hunk 2 above)
Comment (davidgrayston) [reply src/doc_scan_service/session/create/filters/required.supplementary.document.builder.js:13]: This hasn't been added to Java, so perhaps we should leave this out for now?

**Review by echarrod — APPROVED** [review]
👍

**Pull request merged**